main.go

```diff
@@ -456,13 +456,17 @@ func serverListen(context *Context) error {
 	}
 
 	p := proxy.New(
-		tls.NewListener(listener, config),
+		listener,
```

Comment:
I've been pondering this for a bit, and it seems to me that the cleanest way to solve this is to create a new Listener that embeds `tls.Listener` and then overrides `Accept` and returns new server conn with (potentially) modified config (https://golang.org/src/crypto/tls/tls.go?s=1288:1336#L66).

Right now it's not clear how things work, because the mechanism is split over a few places:
- although `listener` could be a `tls.Listener`, it never is
- `ListenWithTLS` really is just a `GetTLSConfig` func
- if `listenTLSConfig` is set, then a conn gets wrapped as a tls.Conn, but there's also a method being called below called `forceHandshake`, which would suggest that the listener actually accepts tls.Conn's in the first place 🤔

Reply:
I like that, overriding Accept is much better than my solution. I'll rework this pull request to incorporate your feedback.